prow/pod-utils/clone/clone.go

```diff
@@ -49,13 +49,14 @@ func Run(refs kube.Refs, dir, gitUserName, gitUserEmail string) Record {
 	commands = append(commands, shellCloneCommand(cloneDir, "git", "fetch", repositoryURL, "--tags", "--prune"))
 	commands = append(commands, shellCloneCommand(cloneDir, "git", "fetch", repositoryURL, refs.BaseRef))
 
-	var checkout string
+	var target string
 	if refs.BaseSHA != "" {
-		checkout = refs.BaseSHA
+		target = refs.BaseSHA
 	} else {
-		checkout = "FETCH_HEAD"
+		target = "FETCH_HEAD"
 	}
-	commands = append(commands, shellCloneCommand(cloneDir, "git", "checkout", checkout))
+	commands = append(commands, shellCloneCommand(cloneDir, "git", "branch", "--force", refs.BaseRef, target))
```

Comment:
One of the assumptions to this tool is that we should handle the case where we're working with a clean slate as well as if we're working on top of an earlier clone, hence the `--force` here in case the branch already existed.